docker-compose-dev.yml

```diff
@@ -0,0 +1,25 @@
+version: '3.4'
```

Comment:
Why is this file added? An old artifact from before I did my own PR where the separate dev compose file was removed? Is this necessary?

Reply:
(to combine my feedback of 'we should just provide production images', if we do that I believe the target added in this file should just be edited in the existing compose file and this file removed)

Reply:
Indeed, it's an old artifact, my bad.

> the target added in this file should just be edited in the existing compose file

If developers have to update their `docker-compose.yml` file to run dev versions, this will not be really practical as it'll be marked as an "unstage change" by git. This might introduce too much friction. However, we can use an env var, defined in `.env`, so people can easily switch to any env.

Reply:
My recommendation is to have `docker-compose.yml` only include what is useful in EVERY environment, and add a `docker-compose.override.yml.dist` file which is designed for environment specific purposes, with defaults for development. Users then copy it to `docker-compose.override.yml` and make any changes necessary. You add the `.dist` to `.gitignore` and then it won't show up as an unstaged change.